midenc-session/src/lib.rs

```diff
@@ -285,13 +285,17 @@ impl Session {
         if self.should_emit(output_type) {
             match self.output_files.path(output_type) {
                 OutputFile::Real(path) => {
-                    let path = if let Some(name) = item.name() {
-                        path.with_file_name(name.as_str())
+                    let file_path = if path.is_dir() {
```

Comment:
The output file path is guaranteed to always be a file path, not a directory, so this check is unnecessary.

Reply:
Hmm, the OutputFile API is designed to always return a file path, so if it's returning a directory there's a somewhere. I'll take a closer look